### ISSUE TITLE
Configurable blobs

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-python#497]
After some discussions, we have decided to change how "external storage" is managed.   Instead of the `external` datatype, we will have `blob-*` datatypes.  For backward compatibility, `external` and `external-*` datatypes will continue to be supported.  

The main advantage is that `blob-*` can be configured to be regular blob datatype, stored in the database or stored in external bulk storage.  So the immediately supported storage protocols will be `table`, `file`, and `s3`.

**Comment by dimitri-yatsenko** [comment]
Here are examples of syntax:
```python
dj.config['store'] = dict(protocol='blob', compress='zlib')
dj.config['store-images'] = dict(protocol='s3', location='/mybucket/store/', secretkey='')
dj.config['store-raw'] = dict(protocol='file', location='/store/', folding=(2,2))
dj.config['store-shared'] = dict(protocol='s3', folding=(1,2))

dj.config['attach'] = dict(store='store')
dj.config['attach-scope'] = dict(store='store-images')
dj.config['blob-images'] = dict(store='store-images')
```


Internally, the DataJoint attribute declaration 
```
field : blob-   # comment
```
is translated into SQL column declaration
```
field longblob NOT NULL COMMENT "blob-:commment"
```

and
```
field : blob-shared
```
into
```sql
field char(51)  COMMENT "blob-shared:"
```
and
```
field : attach-shared
```
into
```sql
field char(51)  COMMENT "attach-shared:comment"
```

**Comment by dimitri-yatsenko** [comment]
Fixed with #532
